### PR TITLE
feat: startknet indexer

### LINKED ISSUE #236: fix the indexer and ensure it works
The landver indexer should be updated to index all emitted events.

### PR DESCRIPTION
**Pull request by rajiabdul**
## Detailed Information

---

## Related Issues

Closes #236 

---

## Type of Change

- [x] 🐛 Bug fix or ⚙️ enhancement
- [ ] ✨ New feature or Chore (change with no new features or fixes)
- [ ] 📚 Documentation update

---

## Checklist (select as many as applicable)

- [x] The code follows the style guidelines of this project.
- [ ] All new and existing tests pass.
- [ ] This pull request is ready to be merged and reviewed.

### CODE REVIEW

File: land-registry-indexer/src/indexer.ts
Comment (fishonamos) [inline land-registry-indexer/src/indexer.ts:1]: Nice. Please separate db operations.

**Comment by fishonamos** [comment]
Nice. Please separate db operations.

File: land-registry-indexer/src/config.ts [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
+  LAND_SOLD: hexToDec("0x0000000000000000000000000000000000000000000000000000000000000009"),
+};
+
+export const createIndexerConfig = () => {
+  const filter = Filter.make({
+    header: "unknown",
+    events: [{
+      address: process.env.LAND_REGISTRY_ADDRESS as `0x${string}`,
+      keys: Object.values(EVENT_KEYS).map((key) => `0x${key.toString()}` as `0x${string}`),
+      transactionStatus: "succeeded",
```
Comment (fishonamos) [inline land-registry-indexer/src/config.ts:31]: Use all status.

File: land-registry-indexer/src/config.ts [hunk 2]
```diff
@@ -1,16 +1,53 @@
...
-  apibaraUrl: process.env.APIBARA_URL || '',
-}; 
+// Helper function to convert hex string to decimal string
+const hexToDec = (hex: string): string => {
+  return BigInt(hex).toString(10);
+};
+
+// Event keys for the Land Registry contract
+export const EVENT_KEYS = {
+  LAND_REGISTERED: hexToDec("0x0000000000000000000000000000000000000000000000000000000000000001"),
```
Comment (fishonamos) [inline land-registry-indexer/src/config.ts:14]: Use the correct event keys. 

https://sepolia.starkscan.co/contract/0x05a4054a1b1389dcd48b650637977280d32f1ad8b3027bc6c7eb606bf7e28bf5#events

File: land-registry-indexer/src/config.ts (hunk 1 above)
Comment (fishonamos) [inline land-registry-indexer/src/config.ts:31]: Use all transaction status here.

**Review by fishonamos — CHANGES_REQUESTED** [review]
Overall great! Please effect the requested changes as soon as possible.

**Comment by rajiabdul** [comment]
Will push my changes now!

**Pull request merged**